plugins/woocommerce-admin/client/marketing/hooks/useIntroductionBanner.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * External dependencies
+ */
+import { useDispatch, useSelect } from '@wordpress/data';
+import { OPTIONS_STORE_NAME } from '@woocommerce/data';
+import { recordEvent } from '@woocommerce/tracks';
+
+type UseIntroductionBanner = {
+	loading: boolean;
+	isIntroductionBannerDismissed: boolean;
+	dismissIntroductionBanner: () => void;
+};
+
+const OPTION_NAME =
+	'woocommerce_marketing_overview_multichannel_banner_dismissed';
+const OPTION_VALUE = 'yes';
+
+export const useIntroductionBanner = (): UseIntroductionBanner => {
+	const { updateOptions } = useDispatch( OPTIONS_STORE_NAME );
+
+	const dismissIntroductionBanner = () => {
+		updateOptions( {
+			[ OPTION_NAME ]: OPTION_VALUE,
+		} );
+		recordEvent( 'marketing_multichannel_banner_dismissed', {} );
+	};
+
+	return useSelect( ( select ) => {
+		const { getOption, isOptionsUpdating, hasFinishedResolution } =
+			select( OPTIONS_STORE_NAME );
+		const isUpdateRequesting = isOptionsUpdating();
+
+		return {
+			loading: ! hasFinishedResolution( 'getOption', [ OPTION_NAME ] ),
+			isIntroductionBannerDismissed:
+				getOption( OPTION_NAME ) === OPTION_VALUE || isUpdateRequesting,
+			dismissIntroductionBanner,
+		};
+	} );
```

Comment:
It would be a bit nicer to specify `deps` for `useSelect` in order to memorize the return.

```suggestion
	}, [] );
```

Reply:
Thanks. I committed your suggestion in [`4925342` (#37110)](https://github.com/woocommerce/woocommerce/pull/37110/commits/49253425b213c97b2a855854055a249817b869dc). Then, I found there is actually a dependency on `dismissIntroductionBanner`, so I made another change in [`1b65be0` (#37110)](https://github.com/woocommerce/woocommerce/pull/37110/commits/1b65be0a5338f3b235f048fa7da29a42330e3227).